microshift.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+export LC_ALL=C.UTF-8
+export LANG=C.UTF-8
+
+source tools.sh
+source snc-library.sh
+
+BUNDLE_TYPE="microshift"
+INSTALL_DIR=crc-tmp-install-data
+CRC_VM_NAME=${CRC_VM_NAME:-crc-microshift}
+BASE_DOMAIN=${CRC_BASE_DOMAIN:-testing}
+MIRROR=${MIRROR:-https://mirror.openshift.com/pub/openshift-v4/$ARCH/clients/ocp-dev-preview}
+OPENSHIFT_RELEASE_VERSION=4.12.0-ec.4
+
+if grep -q -i "release 9" /etc/redhat-release
+then
+  echo "Check if system is registered"
+  # Check the subscription status and register if necessary
+  if ! sudo subscription-manager status >& /dev/null ; then
+     sudo subscription-manager register
+  fi
+else
+  echo "This script only works for RHEL-9"
+fi
+
+run_preflight_checks ${BUNDLE_TYPE}
+rm -fr ${INSTALL_DIR} && mkdir ${INSTALL_DIR}
+
+# Download the oc binary for specific OS environment
+download_oc
+OC=./openshift-clients/linux/oc
+
+# Generate a new ssh keypair for this cluster
+# Create a 521bit ECDSA Key
+rm id_ecdsa_crc* || true
+ssh-keygen -t ecdsa -b 521 -N "" -f id_ecdsa_crc -C "core"
+
+# Configure host (RHEL-9) with required internal repo
+sudo tee /etc/yum.repos.d/latest-fdp.repo >/dev/null <<EOF
+[latest-FDP]
+name = latest-FDP-Buildroot-rpms
+baseurl = http://download.eng.bos.redhat.com/rhel-9/nightly/FDP/latest-FDP-9-RHEL-9/compose/Server/\$basearch/os/
+enabled = 1
+gpgcheck = 0
+EOF
+sudo tee /etc/yum.repos.d/latest-rhaos.repo >/dev/null <<EOF
+[latest-RHAOS]
+name = latest-RHAOS-puddle
+baseurl = http://download.lab.bos.redhat.com/rcm-guest/puddles/RHAOS/plashets/4.12-el9/building/\$basearch/os/
+enabled = 1
+gpgcheck = 0
+EOF
+
+rm -fr microshift
+# Clone microshift repo
+git clone https://github.com/openshift/microshift.git
+pushd microshift
+ssh-keygen -t ecdsa -b 521 -N "" -f id_ecdsa_crc -C "core"
+sed -i '/# customizations/,$d' scripts/image-builder/config/blueprint_v0.0.1.toml
+sed -i 's/redhat/core/g' scripts/image-builder/config/kickstart.ks.template
+sed -i 's/epel-release-latest-8.noarch/epel-release-latest-9.noarch/g' scripts/image-builder/configure.sh
+sed -i 's/rhocp-4.11-for-rhel-8-\${BUILD_ARCH}-rpms/latest-RHAOS/g' scripts/image-builder/build.sh
+sed -i 's/fast-datapath-for-rhel-8-\${BUILD_ARCH}-rpms/latest-FDP/g' scripts/image-builder/build.sh
+scripts/image-builder/configure.sh
+scripts/image-builder/cleanup.sh -full
+make rpm
+packaging/rpm/make-microshift-images-rpm.sh rpm ${OPENSHIFT_PULL_SECRET_PATH} ${ARCH}:${yq_ARCH} rhel-9-${ARCH}
+CONTAINERS_RPM=$(find $(pwd -P)/_output/paack-result -name microshift-containers-\*.$(uname -i).rpm)
+scripts/image-builder/build.sh -pull_secret_file ${OPENSHIFT_PULL_SECRET_PATH} -custom_rpms $CONTAINERS_RPM -authorized_keys_file $(realpath ../id_ecdsa_crc.pub)
```

Comment:
Do you know if this is going to be improved upstream? Feels weird/error-prone to run a command to generate microshift rpms, and then have to manually fetch the list of built rpms and provide it to another script from microshift.

Reply:
@cfergeau I don't know as of now this this is the way to add container images and relative dependencies. It would be better if after GA/TP they just provide a base ISO file which have all the packages and configurations.